### PR TITLE
Bugfixs in Settings Module

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Fix : #1969 

Bugs are elaborated in the below document with the module index as the Settings.
https://docs.google.com/spreadsheets/d/1CB139UvzvUtB1JOHYP0nmEqjxdoHQPGWLGjD_niH9pw/edit#gid=1379931937

### CODE REVIEW

File: tests/Unit/Common/SettingsControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            'website' => 'https://lws.com',
+            'phone' => '9789909887',
+            'address' => 'bangalore',
+            'state' => 'karnataka',
+            'default_currency' => 'USD',
+            'country' => 'IN',
+        ];
+        $v = $this->app['validator']->make($data, $rules);
+        $this->assertTrue($v->passes());
+    }
```
Comment (ajaz-ul-haq) [inline tests/Unit/Common/SettingsControllerTest.php:41]: we should hit an api with invalid data and then assert that response status is 412 as preconditional Failed

File: tests/Unit/Common/SocialMediaControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                'name' => 'required|unique:social_media',
+                'link' => 'required|url|regex:'.$regex,
+            ];
+
+        $data = [
+            'name' => 'Telegram',
+            'link' => 'https://telegram.com',
+        ];
+        $v = $this->app['validator']->make($data, $rules);
+        $this->assertTrue($v->passes());
```
Comment (ajaz-ul-haq) [inline tests/Unit/Common/SocialMediaControllerTest.php:28]: same as above

Comment (sowmiyasubramaniyan) [reply tests/Unit/Common/SocialMediaControllerTest.php]: Done

File: tests/Unit/Common/SettingsControllerTest.php (hunk 1 above)
Comment (sowmiyasubramaniyan) [reply tests/Unit/Common/SettingsControllerTest.php]: Done

**Review by ajaz-ul-haq — DISMISSED** [review]

**Review by RafficMohammed — APPROVED** [review]

**Review by Sheetal34 — CHANGES_REQUESTED** [review]
**Tested with existing db on sever:**
Facing issues there:

- [ ] Admin Panel > System settings 
Select admin panel logo and update. Aadmin panel logo showing on dashboard.
Now click on delete icon and delete admin panel logo. Without clicking on update, admin panel logo will be removed from dashboard.
image

- [ ] Better to give The company name must not be greater than 50 characters. Need to change in tooltip only. Why we are decreasing character size? Before that it's shwoing 50 characters. Now showing 35 characters.
image

- [ ] Create Tax Class > Without selecting tax type, we are successfully able to create tax class

**Review by priyadarshini2101 — APPROVED** [review]
Tested the pr with existing db on server.

All the pointed issues are fixed.

- [ ] Issue with Social media displayed on the client panel, the same social media is shown in the footer and header creating duplicates.

affeb2sm

- [ ]  Admin Panel Logo not showing in the sidebar in Dashboard.

affeb2lo

- [ ] Payment is showing twice in the Stripe description.

- [ ]  The Company name must not be greater than 50 characters.

affebchar

- [ ]  No validation is added in Phone system settings.

affeb2ph

- [ ]  Data is not vanishing even after closing the tab in tax.

affebtx

 

- [ ] Create Tax class without selecting tax type, error message showing outside the tax class page.

affeb2tc

- [ ] The issue with Currency Name. Both are combined and need space between two words.

affeb2cn

**Review by manikz5 — APPROVED** [review]
Tested PR, all mentioned issues are fixed.
After Fix : 
1. Payment is showing twice in the Stripe description.
stripe

2. Issue with Social media displayed on the client panel, the same social media is shown in the footer and header creating duplicates.
socialmedia

3. Admin Panel Logo not showing in the sidebar in Dashboard.
companylogo

4. The Company name must not be greater than 50 characters.
companyname

5. No validation is added in Phone system settings.
phoneNo

6. Tax module : 
taxRatemandatory

7. The issue with Currency Name. Both are combined and need space between two words.
currencyname

**Pull request merged**